### PR TITLE
Added CreateReservation command

### PR DESCRIPTION
**Pull request by lukasz-kkk**

### CODE REVIEW

File: backend/GymBeam/Controllers/ReservationController.cs [hunk 1]
```diff
@@ -39,12 +43,58 @@ public async Task<ActionResult<IEnumerable<ReservationResponse>>> Get()
...
+                StartTime = dto.StartTime
+            };
+            try
+            {
+                if (!Request.Cookies.TryGetValue(Cookies.UserId, out string cookiesUserId))
+                    throw new InvalidCookieException(Cookies.UserId);
+                if (!int.TryParse(cookiesUserId, out userId))
+                    throw new InvalidUserIdException();
+
+                request.UserId = userId;
```
Comment (lukasz-kkk) [inline backend/GymBeam/Controllers/ReservationController.cs:68]: I'm not sure if assigning request.UserId this way is correct

Comment (lukasz-kkk) [inline backend/GymBeam/Controllers/ReservationController.cs:68]: I'm not sure if assigning request.UserId this way is correct

Comment (daopol) [reply backend/GymBeam/Controllers/ReservationController.cs:68]: better to have is from request, maybe admin wants to reserve?

Comment (daopol) [reply backend/GymBeam/Controllers/ReservationController.cs:68]: we need to check if user that sends this post is user that is logged in or admin.
As a user only me or admin can change my reservations.
I would add UserId to reservation request and to reservation command
and add LoggedUserId to only reservation command
So Command will have both UserId(from request) and LoggedUserId(from cookies) but Request will have only UserId.
This way in Command handler you can check whether User is logged user or admin

File: backend/GymBeam/CommandHandlers/ReservationCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+        public Task<Unit> Handle(CreateReservationCommand request, CancellationToken cancellationToken)
+        {
+            var identity = _httpContextAccessor.HttpContext.User.Identity as ClaimsIdentity 
+                ?? throw new AuthenticationFailureException(Resource.ExceptionUserRoleNotFound);
+
+            IEnumerable<Claim> claims = identity.Claims;
+            var roleClaim = claims.FirstOrDefault(c => c.Type == ClaimTypes.Role);
+
+            if (roleClaim.Value != Role.Admin.ToString())
```
Comment (daopol) [inline backend/GymBeam/CommandHandlers/ReservationCommandHandler.cs:29]: you can write it 
if(1condition && 2condition)

File: backend/GymBeam/Controllers/ReservationController.cs [hunk 3]
```diff
@@ -39,12 +43,64 @@ public async Task<ActionResult<IEnumerable<ReservationResponse>>> Get()
...
+                StartTime = dto.StartTime
+            };
+            try
+            {
+                if (!Request.Cookies.TryGetValue(Cookies.UserId, out string cookiesUserId))
+                    throw new InvalidCookieException(Cookies.UserId);
+                if (!int.TryParse(cookiesUserId, out int userIdFromCookies))
+                    throw new InvalidUserIdException();
+
+                request.LoggedUserId = userIdFromCookies;
```
Comment (daopol) [inline backend/GymBeam/Controllers/ReservationController.cs:69]: maybe we can retrive this logged user inside command handler like you did with claim role? If not possible then ok.

File: backend/GymBeam/CommandHandlers/ReservationCommandHandler.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+            var reservation = new Reservation
+            {
+                Activity = activity,
+                User = user,
+                StartTime = request.StartTime
+            };
+
+            _repository.Add(reservation);
+            _repository.SaveChangesAsync();
```
Comment (daopol) [inline backend/GymBeam/CommandHandlers/ReservationCommandHandler.cs:51]: shouldn't this be await? Like everywhere?

**Review by tirey93 — APPROVED** [review]

**Pull request merged**